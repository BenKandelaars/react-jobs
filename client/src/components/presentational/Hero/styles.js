import styled from 'styled-components';
import { globalStyles } from '../globalStyles';
// import rick from './assets/rick.jpg';

export const Wrapper = styled.div`
  height: 100vh;
  min-height: 700px;
  width: 100%;
  font-family: 'Khula', sans-serif;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  overflow: hidden;
  background-color: ${globalStyles.bgColor};
  position: relative;
`;

export const SVGLeft = styled.svg`
  z-index: 0;
  transform: scale(1.5) rotate(90deg);
  margin-bottom: -50px;

  @media (min-width: 600px) {
    transform: scale(2) rotate(90deg);
  }

  @media (min-width: 900px) {
    position: absolute;
    left: -180px;
    top: 15px;
    transform: scale(1.5);
  }
`;

export const SVGRight = styled.svg`
  display: none;

  @media (min-width: 900px) {
    display: inline;
    position: absolute;
    right: -180px;
    top: 15px;
    transform: rotate(180deg) scale(1.5);
  }
`;

export const Article = styled.article`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: auto;
  margin-bottom: auto;
  z-index: 5;
  position: relative;
`;

export const H1 = styled.h1`
  padding: 0;
  margin: 0;
  padding-top: 1rem;

  font-family: 'Khula', sans-serif;
  line-height: 3rem;

  font-size: 10vw;
  color: white;

  @media (min-width: 500px) {
      font-size: 4rem;
  }
`;

export const H2 = styled.h2`
  margin: 10px;
  color: white;
  font-family: 'Open Sans', sans-serif;
  font-weight: 300;
  font-size: 7vw;

  @media (min-width: 500px) {
      font-size: 3rem;
  }
`;

export const P = styled.p`
  color: white;
  text-align: center;

  @media (min-width: 500px) {
    font-size: 1.5rem;
  }
`;

export const P2 = styled.p`
padding: 0 0.3em;
color: white;
text-align: center;
font-size: 1.2rem;

  @media (min-width: 500px) {
    font-size: 1.8rem;
  }
`;

export const Foot = styled.div`
  font-size: 1.5rem;

`;
