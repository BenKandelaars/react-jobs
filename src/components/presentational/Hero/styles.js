import styled from 'styled-components';
import { globalStyles } from '../globalStyles.js';
import rick from "./assets/rick.jpg"
import curlyBraces from "./assets/curlyBraces.svg"

export const Wrapper = styled.div`
  height: 100vh;
  min-height: 700px;
  width: 100%;
  font-family: 'Khula', sans-serif;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  overflow: hidden;
  background-color: ${globalStyles.bgColor};
  position: relative;
`

export const SVGLeft = styled.svg`
  z-index: 0;
  transform: scale(1.5) rotate(90deg);
  margin-bottom: -50px;

  @media (min-width: 600px) {
    transform: scale(2) rotate(90deg);
  }

  @media (min-width: 900px) {
    position: absolute;
    transform: scale(1.5) translate(-250px, 15px);
  }
`

export const SVGRight = styled.svg`
  display: none;

  @media (min-width: 900px) {
    display: inline;
    position: absolute;
    transform: rotate(180deg) scale(1.5) translate(-250px, -15px);
  }
`

export const Article = styled.article`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: auto;
  margin-bottom: auto;
  z-index: 5;
`

export const H1 = styled.h1`
  padding: 0;
  margin: 0;
  padding-top: 1rem;

  font-family: 'Days One', sans-serif;
  line-height: 3rem;

  font-size: 10vw;
  color: white;

  @media (min-width: 500px) {
      font-size: 3rem;
  }
`

export const H2 = styled.h2`
  margin: 0;
  color: white;
  font-family: 'Khula', sans-serif;
  font-size: 7vw;

  @media (min-width: 500px) {
      font-size: 2rem;
  }

`

export const P = styled.p`
  color: white;
  text-align: center;
`

export const Footer = styled.div`
  font-size: 1.5rem;

`
