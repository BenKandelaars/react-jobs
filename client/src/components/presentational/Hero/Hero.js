import React from 'react';
import PropTypes from 'prop-types';
import { Wrapper, H1, H2, P, Article, Foot } from './styles';
import { Button, I, FlexRow } from '../Button/Button';
import { BracesLeft, BracesRight } from './assets/CurlyBraces';

const Centered = props => <Article>{props.children}</Article>;
const Footer = props => <Foot>{props.children}</Foot>;

export function Hero() {
  return (
    <Wrapper>
      <Centered>
        <BracesLeft />
        <BracesRight />
        <H2>Travel. Work.</H2>
        <H1><em>See the world</em></H1>
        <FlexRow>
          <Button>
            <I className="fa fa-search fa-2x" aria-hidden="true" />
            EXPLORE ALL JOBS
          </Button>
          <Button>
            <I className="fa fa-users fa-2x" aria-hidden="true" />
            HIRING? START HERE
          </Button>
        </FlexRow>
        <P>Use Jobbatical to find a tech, business, <br />or creative job abroad.</P>
      </Centered>
      <Footer>
        <P>Currently heaps of teams in loads of <br />countries could use your help! </P>
      </Footer>
    </Wrapper>
  );
}

Centered.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

Footer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};