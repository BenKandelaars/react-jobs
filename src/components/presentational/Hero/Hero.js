import React from 'react';
import { Wrapper, H1, H2, P, Article, Footer } from './styles';
import { Button, I, FlexRow } from '../Button/Button';
import { BracesLeft, BracesRight } from './assets/CurlyBraces';

export function Hero() {
  return (
    <Wrapper>
      <Article>
        <BracesLeft />
        <BracesRight />
        <H2>Travel. Work.</H2>
        <H1><em>See the world</em></H1>
        <FlexRow>
          <Button>
            <I className="fa fa-search fa-2x" aria-hidden="true" />
            EXPLORE ALL JOBS
          </Button>
          <Button>
            <I className="fa fa-users fa-2x" aria-hidden="true" />
            HIRING? START HERE
          </Button>
        </FlexRow>
        <P>Use Jobbatical to find a tech, business, <br />or creative job abroad.</P>
      </Article>
      <Footer>
        <P>Currently heaps of teams in loads of <br />countries could use your help! </P>
      </Footer>
    </Wrapper>
  );
}
