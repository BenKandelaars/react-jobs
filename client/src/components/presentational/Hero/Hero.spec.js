import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Hero } from './Hero';

describe('Hero Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Hero />);
  });

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Hero />, div);
  });

  it('Matches standard output', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Contains one Centered styled tag for the body', () => {
    expect(wrapper.find('Centered').length).toBe(1);
  });

  it('Contains one Footer styled tag for the body', () => {
    expect(wrapper.find('Footer').length).toBe(1);
  });
});
