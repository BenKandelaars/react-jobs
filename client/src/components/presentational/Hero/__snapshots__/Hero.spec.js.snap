// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component matches standard output 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.article>
        <SVG />
        <SVG />
        <styled.h2>
            Travel. Work.
        </styled.h2>
        <styled.h1>
            <em>
                See the world
            </em>
        </styled.h1>
        <styled.div>
            <styled.button>
                <styled.i
                    aria-hidden="true"
                    className="fa fa-search fa-2x"
                />
                EXPLORE ALL JOBS
            </styled.button>
            <styled.button>
                <styled.i
                    aria-hidden="true"
                    className="fa fa-users fa-2x"
                />
                HIRING? START HERE
            </styled.button>
        </styled.div>
        <styled.p>
            Use Jobbatical to find a tech, business, 
            <br />
            or creative job abroad.
        </styled.p>
    </styled.article>
    <styled.div>
        <styled.p>
            Currently heaps of teams in loads of 
            <br />
            countries could use your help! 
        </styled.p>
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.article>
            <SVG />
            <SVG />
            <styled.h2>
                  Travel. Work.
            </styled.h2>
            <styled.h1>
                  <em>
                        See the world
                  </em>
            </styled.h1>
            <styled.div>
                  <styled.button>
                        <styled.i
                              aria-hidden="true"
                              className="fa fa-search fa-2x"
                        />
                        EXPLORE ALL JOBS
                  </styled.button>
                  <styled.button>
                        <styled.i
                              aria-hidden="true"
                              className="fa fa-users fa-2x"
                        />
                        HIRING? START HERE
                  </styled.button>
            </styled.div>
            <styled.p>
                  Use Jobbatical to find a tech, business, 
                  <br />
                  or creative job abroad.
            </styled.p>
      </styled.article>
      <styled.div>
            <styled.p>
                  Currently heaps of teams in loads of 
                  <br />
                  countries could use your help! 
            </styled.p>
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Hero />,
      "_debugID": 49,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.article>
                    <SVG />
                    <SVG />
                    <styled.h2>
                              Travel. Work.
                    </styled.h2>
                    <styled.h1>
                              <em>
                                        See the world
                              </em>
                    </styled.h1>
                    <styled.div>
                              <styled.button>
                                        <styled.i
                                                  aria-hidden="true"
                                                  className="fa fa-search fa-2x"
                                        />
                                        EXPLORE ALL JOBS
                              </styled.button>
                              <styled.button>
                                        <styled.i
                                                  aria-hidden="true"
                                                  className="fa fa-users fa-2x"
                                        />
                                        HIRING? START HERE
                              </styled.button>
                    </styled.div>
                    <styled.p>
                              Use Jobbatical to find a tech, business, 
                              <br />
                              or creative job abroad.
                    </styled.p>
          </styled.article>
          <styled.div>
                    <styled.p>
                              Currently heaps of teams in loads of 
                              <br />
                              countries could use your help! 
                    </styled.p>
          </styled.div>
</styled.div>,
        "_debugID": 50,
        "_renderedOutput": <styled.div>
          <styled.article>
                    <SVG />
                    <SVG />
                    <styled.h2>
                              Travel. Work.
                    </styled.h2>
                    <styled.h1>
                              <em>
                                        See the world
                              </em>
                    </styled.h1>
                    <styled.div>
                              <styled.button>
                                        <styled.i
                                                  aria-hidden="true"
                                                  className="fa fa-search fa-2x"
                                        />
                                        EXPLORE ALL JOBS
                              </styled.button>
                              <styled.button>
                                        <styled.i
                                                  aria-hidden="true"
                                                  className="fa fa-users fa-2x"
                                        />
                                        HIRING? START HERE
                              </styled.button>
                    </styled.div>
                    <styled.p>
                              Use Jobbatical to find a tech, business, 
                              <br />
                              or creative job abroad.
                    </styled.p>
          </styled.article>
          <styled.div>
                    <styled.p>
                              Currently heaps of teams in loads of 
                              <br />
                              countries could use your help! 
                    </styled.p>
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Hero />,
}
`;
